pageserver/src/tenant/timeline.rs

```diff
@@ -926,6 +935,29 @@ impl Timeline {
         self.state.subscribe()
     }
 
+    pub async fn wait_to_become_active(
+        &self,
+        _ctx: &RequestContext, /* Prepare for use by cancellation */
+    ) -> Result<(), TimelineState> {
+        let mut receiver = self.state.subscribe();
+        loop {
+            let current_state = *receiver.borrow_and_update();
+            match current_state {
+                TimelineState::Loading => {
+                    // the sender cannot be dropped because we hold a refence to 'self'
+                    receiver.changed().await.expect("state sender dropped");
```

Comment:
I almost suggested "is this an expect on purpose" :)
```suggestion
                    receiver.changed().await.expect("holding a reference to self");
```